docs/use_bls_wallet_dapp.md

```diff
@@ -0,0 +1,109 @@
+# Use BLS Wallet In Your L2 dApp
+
+This guide will show you how to use BLS Wallet in your L2 dApp (Layer 2 decentralized application) so you can utilize multi-action transactions.
+
+## Download, Install, & Setup Quill
+
+[Quill](../extension/) is a protoype browser extension wallet which intergrates [bls-wallet-clients](../contracts/clients/) to communicate with the BLS Wallet smart contracts & transaction aggregator. It supports most of the functionality in [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193).
+
+Currently, we have the contracts deployed to the networks/chains listed [here](https://github.com/web3well/bls-wallet/tree/main/contracts/networks). If your desired networks isn't there, you can use the [Remote Development](./remote_development.md) contract deployment intrsuctions or request a network deploy by [opening an issue](https://github.com/web3well/bls-wallet/issues/new) or [starting a discussion](https://github.com/web3well/bls-wallet/discussions/new).
+
+Below are the instructions for 2 ways you can add Quill to your browser.
+
+### Prebuilt
+
+Go to the [releases page](https://github.com/web3well/bls-wallet/releases) and scroll down to the latest release. In the `Assets` section, download the extension for either Chrome, Firefox, or Opera. To install, simply drag and drop the file into your browser on the extensions page or follow instructions for installing extensions from a file for your browser.
+
+### From Repo
+
+Follow the instructions in either [Local Development](./local_development.md) or [Remote Development](./remote_development.md) to setup this repo and install Quill.
+
+### Setup Quill
+
+Folow the setup instructions when ou first open Quill, and then create a wallet if there is not one there already.
+
+## Connect Your dApp to Quill
+
+Next, connect your dApp to Quill just like you would any other extension wallet.
```

Comment:
A lot of apps use custom connection libraries like [web3modal (usage)](https://github.com/WalletConnect/web3modal#usage) and [rainbow-connect (usage)](https://rainbowkit.vercel.app/docs/connect-button). 
Can add a small point/link about connecting using these libs

Reply:
Added